tests/unit/s2n_client_hello_recv_test.c

```diff
@@ -102,7 +102,7 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_stuffer_write(&server_conn->handshake.io, &client_hello));
         EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
 
-        EXPECT_EQUAL(server_conn->server_protocol_version, i == 0 ? S2N_TLS12 : S2N_TLS13);
+        EXPECT_EQUAL(server_conn->server_protocol_version, S2N_TLS12);
```

Comment:
`conn->server_protocol_version` is set to `s2n_highest_protocol_version` in `s2n_connection_new`. `s2n_highest_protocol_version` is either TLS 1.2 or TLS 1.3, depending on whether `s2n_disable/enable_tls13_test` functions have been used.
### Previously
server_protocol_version was never modified for SSLv2 client hellos. This line was just asserting on `s2n_highest_protocol_version` which was toggled between test runs by `s2n_disable/enable_tls13_test`.

### Proposed
`server_protocol_version` is now modified by this section in `process_client_hello`.
https://github.com/aws/s2n-tls/blob/cc2ea721e54d90e11212d9b02e9fb297aeaa273d/tls/s2n_client_hello.c#L535-L538

From my reading/audit of the codebase, `server_protocol_version` is mostly used for various checks and assertions related to downgrade protection. From the brief audit that I did, I didn't see any places that this would be a noticeable behavior change, but this part of the codebase is pretty new to me and our SSLV2 test coverage is pretty low ☹️ 

I'll also throw out that `server_protocol_version` being set to TLS13 while using SSLV2 client hellos seems suspicious given this RFC requirement
> Implementations MUST NOT negotiate TLS 1.3 or later using an SSL version 2.0 compatible CLIENT-HELLO.
https://www.rfc-editor.org/rfc/rfc8446#appendix-D.5





> Note: I think the code chunk above should just always be assigning `S2N_TLS12`. When I made that change, all the test cases continued to pass exception one of the renegotiate ones. Need to investigate further